### PR TITLE
Fix #3859: Cleanup private linker library

### LINKED ISSUE #3859: Cleanup linker library
- Move it into `org.scalajs` package.
- Run the IR cleaner on it.

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by sjrd** [comment]
The first commit of this PR will increase the fastOpt size due to the longer (and uncompressed) name for `RuntimeLong`. Now that the compression is an implementation detail of the emitter, we should take the opportunity to compress `org.scalajs.linker.runtime.RuntimeLong`, for example to just `RTLong` (so for example it would end up as `$c_RTLong`).

File: project/JavalibIRCleaner.scala [hunk 1]
```diff
@@ -31,19 +31,12 @@ import sbt.{Logger, MessageOnlyException}
  *    the JS load spec of the mentioned class ref.
  *
  *  Afterwards, we check that the IR does not contain any reference to classes
- *  undef the `scala.*` package, with the unfortunate exception of
- *  `scala.scalajs.runtime.UndefinedBehaviorError`.
+ *  undef the `scala.*` package.
```
Comment (sjrd) [inline project/JavalibIRCleaner.scala]: While you're there, fix the typo of `under`?

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/JSGen.scala [hunk 2]
```diff
@@ -76,13 +76,15 @@ private[emitter] final class JSGen(val semantics: Semantics,
...
-     * genName(ClassName).
+    /* Fill the cache with the compressed form of java.lang.Object,
+     * java.lang.String and org.scalajs.linker.runtime.RuntimeLong, so that we
+     * do not have to take care of them in genName(ClassName).
      */
     val cache = mutable.Map.empty[ClassName, String]
     cache.put(ObjectClass, "O")
     cache.put(BoxedStringClass, "T")
+    cache.put(LongImpl.RuntimeLongClass, "RTLong")
+    cache.put(LongImpl.RuntimeLongModuleClass, "RTLong$")
```
Comment (sjrd) [inline linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/JSGen.scala]: This is new.

File: ci/checksizes.sh [hunk 3]
```diff
@@ -34,20 +34,20 @@ REVERSI_OPT_GZ_SIZE=$(stat '-c%s' "$REVERSI_OPT.gz")
...
+    REVERSI_PREOPT_EXPECTEDSIZE=409000
     REVERSI_OPT_EXPECTEDSIZE=90000
     REVERSI_PREOPT_GZ_EXPECTEDSIZE=54000
     REVERSI_OPT_GZ_EXPECTEDSIZE=25000
     ;;
   2.13.1)
-    REVERSI_PREOPT_EXPECTEDSIZE=557000
-    REVERSI_OPT_EXPECTEDSIZE=123000
+    REVERSI_PREOPT_EXPECTEDSIZE=554000
+    REVERSI_OPT_EXPECTEDSIZE=122000
```
Comment (sjrd) [inline ci/checksizes.sh]: The changes in checksizes are new, with improved sizes (whereas before fastOpt was increasing by about 5KB).

File: linker-private-library/src/main/scala/org/scalajs/linker/runtime/RuntimeLong.scala [hunk 4]
```diff
@@ -535,10 +535,7 @@ final class RuntimeLong(val lo: Int, val hi: Int)
 
 }
 
-/* `extends java.io.Serializable` prevents scalac from inserting
- * `extends scala.Serializable`.
- */
-object RuntimeLong extends java.io.Serializable {
+object RuntimeLong {
```
Comment (sjrd) [inline linker-private-library/src/main/scala/org/scalajs/linker/runtime/RuntimeLong.scala]: This is new. See the additional paragraph in the commit message.

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/JSGen.scala (hunk 2 above)
Comment (gzm0) [reply linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/JSGen.scala]: We do not want this for UBE?

Comment (sjrd) [reply linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/JSGen.scala]: We have basically a fixed number of occurrences of UBE, for its definition and the throwing helper function. Whereas `RuntimeLong` appears very often in the code that uses longs.

**Comment by gzm0** [comment]
LGTM

**Pull request merged**